### PR TITLE
TOOLS-2645: Check for duplicate index keys after converting legacy in…

### PR DESCRIPTION
**Pull request by huan-Mongo**
…dex definitions
This PR depends on https://github.com/mongodb/mongo-tools-common/pull/45/files . I will revendor after the MTC PR gets merged. 
Here I introduced flag fixDuplicatedLegacyIndexes. When enabled, after legacy indexes get converted, the indexes with duplicated index key will be skipped.

### CODE REVIEW

File: mongorestore/options.go [hunk 1]
```diff
@@ -67,23 +67,24 @@ func (*InputOptions) Name() string {
...
+	NumParallelCollectionsOption     = "--numParallelCollections"
+	NumInsertionWorkersOption        = "--numInsertionWorkersPerCollection"
+	StopOnErrorOption                = "--stopOnError"
+	BypassDocumentValidationOption   = "--bypassDocumentValidation"
+	PreserveUUIDOption               = "--preserveUUID"
+	TempUsersCollOption              = "--tempUsersColl"
+	TempRolesCollOption              = "--tempRolesColl"
+	BulkBufferSizeOption             = "--batchSize"
+	FixDottedHashedIndexesOption     = "--fixDottedHashIndex"
+	FixDuplicatedLegacyIndexesOption = "--fixDuplicatedLegacyIndexes"
```
Comment (rychipman) [inline mongorestore/options.go:87]: I may be recalling an outdated discussion, but I thought when we last discussed we were not going to add a new flag and instead make it part of the default `--convertLegacyIndexes` behavior

File: mongorestore/restore.go [hunk 2]
```diff
@@ -306,17 +306,27 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument, ns string) {
+func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument, ns string) []IndexDocument {
+	indexKeys := make(map[string]bool) // A set contains all the unique index's json strings
```
Comment (rychipman) [inline mongorestore/restore.go:310]: small nit, but the go idiom is to use `map[string]struct{}` for sets

File: mongorestore/options.go (hunk 1 above)
Comment (huan-Mongo) [reply mongorestore/options.go:87]: I may remember incorrectly, @tfogo Do you remembre if we would need a flag or not?

File: mongorestore/restore.go (hunk 2 above)
Comment (huan-Mongo) [reply mongorestore/restore.go:310]: Good idea!

File: mongorestore/options.go (hunk 1 above)
Comment (tfogo) [reply mongorestore/options.go:87]: I think we agreed on not having a flag but adding a note to the documentation.

Comment (huan-Mongo) [reply mongorestore/options.go:87]: ah, cool. I will remove the flag then. Thanks!

File: mongorestore/mongorestore_test.go [hunk 3]
```diff
@@ -770,6 +770,73 @@ func TestDeprecatedIndexOptions(t *testing.T) {
...
+	}
+	Convey("With a test MongoRestore", t, func() {
+		args := []string{
+			ConvertLegacyIndexesOption,
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		Convey("Creating index with invalid option and --convertLegacyIndexes should succeed", func() {
```
[the remark this answers is not in the document]
Comment (dragonlee8) [reply mongorestore/mongorestore_test.go]: Good catch! fixed.

File: mongorestore/restore.go [hunk 4]
```diff
@@ -306,17 +306,28 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
...
+func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument, ns string) []IndexDocument {
+	indexKeys := make(map[string]struct{}) // A set contains all the unique index's json strings
+	var indexesConverted []IndexDocument
+	var exists = struct{}{}
 	for _, index := range indexes {
 		bsonutil.ConvertLegacyIndexKeys(index.Key, ns)
+		indexString := bsonutil.CreateExtJSONString(index.Key)
+		if _, ok := indexKeys[indexString]; ok {
+			// skip duplicated indexes
+			continue
```
[the remark this answers is not in the document]
Comment (huan-Mongo) [reply mongorestore/restore.go:319]: Make sense, added the log

File: mongorestore/mongorestore_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (huan-Mongo) [reply mongorestore/mongorestore_test.go]: Good catch. fixed!

**Pull request merged**